app/controllers/steps/capital/savings_controller.rb

```diff
@@ -13,6 +13,21 @@ def update
         )
       end
 
+      def destroy
+        saving_record.destroy
+
+        if savings.reload.any?
+          redirect_to edit_steps_capital_savings_summary_path, success: t('.success_flash')
+        else
+          # If this was the last remaining record, redirect to the saving type page
+          redirect_to edit_steps_capital_saving_type_path, success: t('.success_flash')
+        end
+      end
+
+      def confirm_destroy
+        @saving = helpers.present(saving_record)
+      end
```

Comment:
Not sure what is happening here a comment may be helpful.

Reply:
I think it's just setting the saving that will be displayed on the confirm destroy page. We have a savings presenter that will present some of the fields that need formatting but I haven't done that in this PR so I've removed for now and will add back and add a comment for the next PR.